src/app/tests/TestCommissioningWindowManager.cpp

```diff
@@ -406,4 +409,128 @@ TEST_F(TestCommissioningWindowManager, TestCheckCommissioningWindowManagerEnhanc
     chip::DeviceLayer::PlatformMgr().RunEventLoop();
 }
 
+chip::DeviceLayer::ChipDeviceEvent CreateEvent(uint16_t eventType)
+{
+    chip::DeviceLayer::ChipDeviceEvent event;
+    event.Type = eventType;
+    return event;
+}
+
+// Verify that the commissioning window is closed when commissioning has completed
+TEST_F(TestCommissioningWindowManager, TestOnPlatformEventCommissioningComplete)
+{
+    CommissioningWindowManager & commissionMgr = Server::GetInstance().GetCommissioningWindowManager();
+
+    EXPECT_EQ(commissionMgr.OpenBasicCommissioningWindow(commissionMgr.MaxCommissioningTimeout(),
+                                                         CommissioningWindowAdvertisement::kDnssdOnly),
+              CHIP_NO_ERROR);
+    EXPECT_TRUE(commissionMgr.IsCommissioningWindowOpen());
+
+    auto event = CreateEvent(chip::DeviceLayer::DeviceEventType::kCommissioningComplete);
+
+    commissionMgr.OnPlatformEvent(&event);
+    // When the commissioning has completed (kCommissioningComplete event) OnPlatformEvent cleans up, closes active sessions and the
+    // commissioning window is closed The device should no longer be discoverable or accept new commissioners
+    EXPECT_FALSE(commissionMgr.IsCommissioningWindowOpen());
+
+// When BLE is enabled
+#if CONFIG_NETWORK_LAYER_BLE && CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION
+    EXPECT_FALSE(Server::GetInstance().GetBleLayerObject()->IsBleClosing());
+#endif
+}
+
+// Verify that the when fail-safe timer expires commissioning mode is disabled
+TEST_F(TestCommissioningWindowManager, TestOnPlatformEventFailSafeTimerExpired)
+{
+    CommissioningWindowManager & commissionMgr = Server::GetInstance().GetCommissioningWindowManager();
+
+    EXPECT_EQ(commissionMgr.OpenBasicCommissioningWindow(commissionMgr.MaxCommissioningTimeout(),
+                                                         CommissioningWindowAdvertisement::kDnssdOnly),
+              CHIP_NO_ERROR);
+
+    auto event = CreateEvent(chip::DeviceLayer::DeviceEventType::kFailSafeTimerExpired);
+
+    commissionMgr.OnPlatformEvent(&event);
+    EXPECT_TRUE(commissionMgr.IsCommissioningWindowOpen());
```

Comment:
Wait, so the commissioning window manager doesn't auto-close the commissioning window on failsafe expiry?